### ISSUE TITLE
bower doesn't have permission to create folder (in home directory) by default

### ISSUE THREAD

**Issue by Hubcapp** [Athou/commafeed#704]
Hey again. My virtual machine exploded when the physical machine under it (2005 spare desktop) also exploded. I have a backup from a few months ago, but I decided it was probably a good time to start fresh anyway.

So, I installed a fresh Debian Jessie and tried to compile commafeed again. 

Here's the full error log of what happened after following the README. I'll provide a short summary of the useful information after each error log.

First downloaded commafeed using git command from the README into the user's home directory.

Ran other README commands to satisfy dependencies.

Ran command:
```mvn clean package```
as user in ~/commafeed
With result:
```
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building CommaFeed 2.2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ commafeed ---
[INFO] 
[INFO] --- git-commit-id-plugin:2.1.11:revision (default) @ commafeed ---
[INFO] 
[INFO] --- frontend-maven-plugin:0.0.20:install-node-and-npm (install node and npm) @ commafeed ---
[INFO] Node v0.10.35 is already installed.
[INFO] Found NPM version 2.2.0
[INFO] 
[INFO] --- frontend-maven-plugin:0.0.20:npm (npm install) @ commafeed ---
[INFO] Running 'npm install --color=false' in /home/user/commafeed
[INFO] 
[INFO] --- frontend-maven-plugin:0.0.20:bower (bower install) @ commafeed ---
[INFO] Running 'bower install' in /home/user/commafeed
[INFO] bower readabilicons#34c55561c5b8ec6e90714b50237c06b13cb9d59c           cached git://github.com/arc90/readability-readabilicons.git#34c55561c5
[INFO] bower readabilicons#34c55561c5b8ec6e90714b50237c06b13cb9d59c         validate 34c55561c5 against git://github.com/arc90/readability-readabilicons.git#34c55561c5b8ec6e90714b50237c06b13cb9d59c
[INFO] bower jquery#2.1.1                                                     cached git://github.com/jquery/jquery.git#2.1.1
[INFO] bower jquery#2.1.1                                                   validate 2.1.1 against git://github.com/jquery/jquery.git#2.1.1
[INFO] bower jquery-ui#1.10.3                                                 cached git://github.com/components/jqueryui.git#1.10.3
[INFO] bower jquery-ui#1.10.3                                               validate 1.10.3 against git://github.com/components/jqueryui.git#1.10.3
[INFO] bower jquery-mousewheel#3.1.12                                         cached git://github.com/jquery/jquery-mousewheel.git#3.1.12
[INFO] bower jquery-mousewheel#3.1.12                                       validate 3.1.12 against git://github.com/jquery/jquery-mousewheel.git#3.1.12
[INFO] bower lodash#2.4.1                                                     cached git://github.com/lodash/lodash.git#2.4.1
[INFO] bower lodash#2.4.1                                                   validate 2.4.1 against git://github.com/lodash/lodash.git#2.4.1
[INFO] bower bootstrap#3.3.1                                                  cached git://github.com/twbs/bootstrap.git#3.3.1
[INFO] bower bootstrap#3.3.1                                                validate 3.3.1 against git://github.com/twbs/bootstrap.git#3.3.1
[INFO] bower font-awesome#3.2.1                                               cached git://github.com/FortAwesome/Font-Awesome.git#3.2.1
[INFO] bower font-awesome#3.2.1                                             validate 3.2.1 against git://github.com/FortAwesome/Font-Awesome.git#3.2.1
[INFO] bower angular#1.3.6                                                    cached git://github.com/angular/bower-angular.git#1.3.6
[INFO] bower angular#1.3.6                                                  validate 1.3.6 against git://github.com/angular/bower-angular.git#1.3.6
[INFO] bower angular-resource#1.3.6                                           cached git://github.com/angular/bower-angular-resource.git#1.3.6
[INFO] bower angular-resource#1.3.6                                         validate 1.3.6 against git://github.com/angular/bower-angular-resource.git#1.3.6
[INFO] bower angular-route#1.3.6                                              cached git://github.com/angular/bower-angular-route.git#1.3.6
[INFO] bower angular-route#1.3.6                                            validate 1.3.6 against git://github.com/angular/bower-angular-route.git#1.3.6
[INFO] bower angular-sanitize#1.3.6                                           cached git://github.com/angular/bower-angular-sanitize.git#1.3.6
[INFO] bower angular-sanitize#1.3.6                                         validate 1.3.6 against git://github.com/angular/bower-angular-sanitize.git#1.3.6
[INFO] bower angular-touch#1.3.6                                              cached git://github.com/angular/bower-angular-touch.git#1.3.6
[INFO] bower angular-touch#1.3.6                                            validate 1.3.6 against git://github.com/angular/bower-angular-touch.git#1.3.6
[INFO] bower angular-animate#1.3.6                                            cached git://github.com/angular/bower-angular-animate.git#1.3.6
[INFO] bower angular-animate#1.3.6                                          validate 1.3.6 against git://github.com/angular/bower-angular-animate.git#1.3.6
[INFO] bower angular-ui-router#0.2.13                                         cached git://github.com/angular-ui/ui-router.git#0.2.13
[INFO] bower angular-ui-router#0.2.13                                       validate 0.2.13 against git://github.com/angular-ui/ui-router.git#0.2.13
[INFO] bower angular-ui-utils#0.1.0                                           cached git://github.com/angular-ui/ui-utils.git#0.1.0
[INFO] bower angular-ui-utils#0.1.0                                         validate 0.1.0 against git://github.com/angular-ui/ui-utils.git#0.1.0
[INFO] bower angular-ui-select2#0.0.5                                         cached git://github.com/angular-ui/ui-select2.git#0.0.5
[INFO] bower angular-ui-select2#0.0.5                                       validate 0.0.5 against git://github.com/angular-ui/ui-select2.git#0.0.5
[INFO] bower angular-bootstrap#0.2.0                                          cached git://github.com/angular-ui/bootstrap-bower.git#0.2.0
[INFO] bower angular-bootstrap#0.2.0                                        validate 0.2.0 against git://github.com/angular-ui/bootstrap-bower.git#0.2.0
[INFO] bower angular-loading-bar#0.6.0                                        cached git://github.com/chieffancypants/angular-loading-bar.git#0.6.0
[INFO] bower angular-loading-bar#0.6.0                                      validate 0.6.0 against git://github.com/chieffancypants/angular-loading-bar.git#0.6.0
[INFO] bower angular-translate#2.5.2                                          cached git://github.com/PascalPrecht/bower-angular-translate.git#2.5.2
[INFO] bower angular-translate#2.5.2                                        validate 2.5.2 against git://github.com/PascalPrecht/bower-angular-translate.git#2.5.2
[INFO] bower angular-translate-loader-static-files#2.5.2                      cached git://github.com/PascalPrecht/bower-angular-translate-loader-static-files.git#2.5.2
[INFO] bower angular-translate-loader-static-files#2.5.2                    validate 2.5.2 against git://github.com/PascalPrecht/bower-angular-translate-loader-static-files.git#2.5.2
[INFO] bower ngInfiniteScroll#1.0.0                                           cached git://github.com/sroze/ngInfiniteScroll.git#1.0.0
[INFO] bower ngInfiniteScroll#1.0.0                                         validate 1.0.0 against git://github.com/sroze/ngInfiniteScroll.git#1.0.0
[INFO] bower ng-grid#2.0.6                                                    cached git://github.com/angular-ui/ng-grid.git#2.0.6
[INFO] bower ng-grid#2.0.6                                                  validate 2.0.6 against git://github.com/angular-ui/ng-grid.git#2.0.6
[INFO] bower mousetrap#1.4.6                                                  cached git://github.com/ccampbell/mousetrap.git#1.4.6
[INFO] bower mousetrap#1.4.6                                                validate 1.4.6 against git://github.com/ccampbell/mousetrap.git#1.4.6
[INFO] bower momentjs#2.8.4                                                   cached git://github.com/moment/moment.git#2.8.4
[INFO] bower momentjs#2.8.4                                                 validate 2.8.4 against git://github.com/moment/moment.git#2.8.4
[INFO] bower devicejs#0.1.16                                                  cached git://github.com/matthewhudson/device.js.git#0.1.16
[INFO] bower devicejs#0.1.16                                                validate 0.1.16 against git://github.com/matthewhudson/device.js.git#0.1.16
[INFO] bower zocial-less#1.0.0                                                cached git://github.com/gustavohenke/zocial-less.git#1.0.0
[INFO] bower zocial-less#1.0.0                                              validate 1.0.0 against git://github.com/gustavohenke/zocial-less.git#1.0.0
[INFO] bower swagger-ui#2.0.24                                                cached git://github.com/wordnik/swagger-ui.git#2.0.24
[INFO] bower swagger-ui#2.0.24                                              validate 2.0.24 against git://github.com/wordnik/swagger-ui.git#2.0.24
[INFO] bower select2#~3.4                                                     cached git://github.com/ivaynberg/select2.git#3.4.8
[INFO] bower select2#~3.4                                                   validate 3.4.8 against git://github.com/ivaynberg/select2.git#~3.4
[INFO] bower angular                                                extra-resolution Unnecessary resolution: angular#1.3.6
[INFO] bower angular-translate                                      extra-resolution Unnecessary resolution: angular-translate#2.5.2
[INFO] bower                                                                  EACCES EACCES, mkdir '/home/user/commafeed/src/main/app/lib'
[INFO] 
[INFO] Stack trace:
[INFO] Error: EACCES, mkdir '/home/user/commafeed/src/main/app/lib'
[INFO] 
[INFO] Console trace:
[INFO] Trace
[INFO]     at StandardRenderer.error (/home/user/commafeed/node_modules/bower/lib/renderers/StandardRenderer.js:82:17)
[INFO]     at Logger.<anonymous> (/home/user/commafeed/node_modules/bower/bin/bower:110:22)
[INFO]     at Logger.emit (events.js:95:17)
[INFO]     at Logger.emit (/home/user/commafeed/node_modules/bower/node_modules/bower-logger/lib/Logger.js:29:39)
[INFO]     at /home/user/commafeed/node_modules/bower/lib/commands/index.js:40:20
[INFO]     at _rejected (/home/user/commafeed/node_modules/bower/node_modules/q/q.js:797:24)
[INFO]     at /home/user/commafeed/node_modules/bower/node_modules/q/q.js:823:30
[INFO]     at Promise.when (/home/user/commafeed/node_modules/bower/node_modules/q/q.js:1035:31)
[INFO]     at Promise.promise.promiseDispatch (/home/user/commafeed/node_modules/bower/node_modules/q/q.js:741:41)
[INFO]     at /home/user/commafeed/node_modules/bower/node_modules/q/q.js:557:44
[INFO] 
[INFO] System info:
[INFO] Bower version: 1.3.12
[INFO] Node version: 0.10.35
[INFO] OS: Linux 3.2.0-4-amd64 x64
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 12.928s
[INFO] Finished at: Thu Feb 12 14:58:45 EST 2015
[INFO] Final Memory: 18M/44M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.20:bower (bower install) on project commafeed: Failed to run task: 'bower install' failed. (error code 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

It says that bower wasn't able to create the folder it wanted to, which usually means "permission error". 

So, I decide to run 
```sudo mvn clean package```
With result:
```
[INFO] Scanning for projects...
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building CommaFeed 2.2.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ commafeed ---
[INFO] 
[INFO] --- git-commit-id-plugin:2.1.11:revision (default) @ commafeed ---
[INFO] 
[INFO] --- frontend-maven-plugin:0.0.20:install-node-and-npm (install node and npm) @ commafeed ---
[INFO] Node v0.10.35 is already installed.
[INFO] Found NPM version 2.2.0
[INFO] 
[INFO] --- frontend-maven-plugin:0.0.20:npm (npm install) @ commafeed ---
[INFO] Running 'npm install --color=false' in /home/user/commafeed
[INFO] 
[INFO] --- frontend-maven-plugin:0.0.20:bower (bower install) @ commafeed ---
[INFO] Running 'bower install' in /home/user/commafeed
[INFO] bower ESUDO         Cannot be run with sudo
[INFO] 
[INFO] Additional error details:
[INFO] Since bower is a user command, there is no need to execute it with superuser permissions.
[INFO] If you're having permission errors when using bower without sudo, please spend a few minutes learning more about how your system should work and make any necessary repairs.
[INFO] 
[INFO] http://www.joyent.com/blog/installing-node-and-npm
[INFO] https://gist.github.com/isaacs/579814
[INFO] 
[INFO] You can however run a command with sudo using --allow-root option
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 8.653s
[INFO] Finished at: Thu Feb 12 15:03:10 EST 2015
[INFO] Final Memory: 18M/44M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.20:bower (bower install) on project commafeed: Failed to run task: 'bower install' failed. (error code 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```
Basically just a snide message telling me that running bower as root is wrong, and that I must not know much about Linux. Their "--allow root" option doesn't work since I'm not running bower directly.

So, I break out
```chmod 777 ~/commafeed -R```
at which point, of course, bower is allowed to make the folder.

Needless to say, I didn't try to sabotage myself by setting the permissions so that compiling would fail. I just ran ```git clone https://github.com/Athou/commafeed.git``` in ~, and that's what happened.

**Comment by Athou** [comment]
Really strange, bower should not need special permissions to write in that directory. What were the permissions for that directory before chmod 777 ?

**Comment by Athou** [comment]
I started a new virtual machine with a fresh debian 7 but I couldn't reproduce the problem.

**Comment by Hubcapp** [comment]
I'll try to reproduce and give steps in about a week or so, but if it can't be reproduced, I guess just close the ticket.

**Comment by Hubcapp** [comment]
I'm still not really sure why this happened, but I got the error again; this time in /opt. All the errors just went away after I tried compiling in ~/lordPleaseHelp/. I dunno, just make sure the folder you're in has write permissions for normal users, I guess.
